Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
+    var maxConnections: Int
+    var time: Int
+    
+    public typealias ConnectionFactory = () throws -> Connection
+    
+    init(makeConnection: @escaping ConnectionFactory, maxConnections: Int) {
+        self.makeConnection = makeConnection
+        self.maxConnections = maxConnections
+        storage = [:]
+        time = 0
+    }
+    
+    static var threadId: pthread_t {
+        return pthread_self()
+    }
+    
+    func connection() throws -> Connection {
+        if let threadConnection = storage[ThreadConnectionPool.threadId] {
+            return threadConnection.conn
```

Comment:
We are not removing the connection from the pool. This will make it "available" for other threads to use, which could have issues with "last_insert_id()" and prevent transactions from working properly.